fontforge/featurefile.c

```diff
@@ -7304,17 +7329,29 @@ static void CopySplineFontGroupsForFeatureFile(SplineFont *sf, struct parseState
   }
 }
 
+static bool fea_isInUFO(char *filename) {
```

Comment:
I understand what this is doing but don't have a strong sense how "safe" it is.

Reply:
From what I can tell, it's completely safe. If someone tries to import a fea file that's part of a foreign UFO, for example, they'd still want the imports to be resolved as they would be if a fea file is part of a UFO.

Reply:
Now that I think about this again, can we not just compare that this filename is contained by `sf->filename` (possibly applying `GFileGetAbsoluteName`)?

Reply:
I don't think so, because of  it's valid to have non-`features.fea` `.fea` files in a UFO. You'd at least need to check that the file is named `features.fea`.

Reply:
Right, but I guess if you checked that + against the filename that's a stronger check than some random path that may or may not come from the same ufo

Reply:
Wait, what? I don't understand what's wrong with the way I'm doing it. Who cares if the file is contained by `sf->filename`? That could e.g. be the case with `sfdir`, the directory version of SFD. I think also those weird "Apple font suitcases" we saw discussed elsewhere could also apply here. Just checking that the path ends with `UFO[23]?` is much simpler than assuring we're "contained" by `sf->filename`, but then also making sure that it's not a false positive (we're not in a font suitcase or in an SFD or any other way I'm not thinking of which could trigger true).

Also, I consider the fact that we're not making sure that `features.fea` comes from the same UFO a feature and not a bug. If the `features.fea` is part of any UFO, than obviously `include` resolution should happen exactly as in the UFO spec. Imagine a variable font with 16 masters…if you include a `features.fea` inside a foreign UFO, which way do you want `include` to resolve? Obviously the same as it would in that UFO.

So yeah, not convinced.

Reply:
Ok, fair.